starknet_py/net/account/account.py

```diff
@@ -95,14 +99,13 @@ def __init__(
                 raise ValueError(
                     "Either a signer or a key_pair must be provided in Account constructor."
                 )
-            if chain is None:
+            if self._chain_id is None:
```

Comment:
Maybe let's fetch chain id if it's `None`?

Reply:
No strong opinion, on the other hand, is ok to demand users, to explicitly set things like this, It might protect from some unconscious actions

Reply:
In my opinion it makes sense to allow passing it, but always fetch it if not passed. At the moment I don't see any potential issues that it may cause.